pkg/env/env.go

```diff
@@ -323,13 +361,19 @@ func (e *testEnv) Run(m *testing.M) int {
 		panic("context not set") // something is terribly wrong.
 	}
 
+	if !e.gatherMode {
```

Comment:
Might be a silly question, is this stats will be collected by default eventhogh developer is not explicitely enabling gatherMode var?
If so I am just curious to know the necessity of gatherStats().

Reply:
The current way I am trying to gather stats is a very hacky on at the best since I wanted to retain the compatibility of the APIs intact and did not want to introduce a new top level construct. 

So, what I do is, when the `Run` method is called, we set the gather mode and run the tests in gather mode. Which basically aggregates the test results and do a recursive call to the `Run` again but this time with `gatherMode = false` as the deferred operation. This ensures the tests get executed now in the right context. 

But you are right that I don't really need the `gatherStats` method to do the same. It started with a different thought in my mind when I introduced this function and did not cleanup yet.  I can achieve this entire flow without actually having that function as well.